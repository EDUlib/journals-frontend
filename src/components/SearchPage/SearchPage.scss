@media (min-width: 992px){
  .search-filter-panel>div{
    margin: 20px 16px 0px 16px;
  }
}

@media (max-width: 991px){
  .search-filter-panel{
    justify-content: space-between;

    >div{
        margin: 0px 5px 0px 5px;
    }
  }
}

.search-filter-panel {
  display: flex;
  border-bottom: 1px solid $light-grey;
  flex-wrap: wrap;
}

.search-filter-panel > div {
  text-align: center;
}

.btn.filter-btn{
  line-height: 1;
  padding: 0 0 8px 0;
  color:  $slate-grey;
  background-color: $white;
  border: none;
  border-radius: 0px;
  font-size: 14px;

  &:hover{
    cursor: pointer;
  }

  &:focus{
    box-shadow: none;
    color: $primary;
    font-weight: bold;
  }
}

.btn.filter-btn-selected{
  color: $primary;
  border-bottom: 2px solid $primary;
  font-weight: bold;
}
