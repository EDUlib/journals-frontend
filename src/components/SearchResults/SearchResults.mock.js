import {
  FILTER_ID_ALL, FILTER_ID_DOCUMENTS, FILTER_ID_IMAGES, FILTER_ID_VIDEOS,
} from '../../data/constants/filterTypes';

const mockSearchResults = {
  meta: {
    total_count: 3,
    text_count: 0,
    image_count: 1,
    video_count: 1,
    doc_count: 1,
  },
  hits: [
    {
      block_id: 15,
      block_title: 'block 1',
      block_type: 'pdf',
      breadcrumbs: ['x', 'y'],
      highlights: [
        '1 <b>foo </b>',
        '2 <b>foo </b>',
        '3 <b>foo </b>',
        '4 <b>foo </b>',
        '5 <b>foo </b>',
      ],
      journal_about_page_id: 12,
      journal_id: 5,
      journal_name: 'new journal',
      page_id: 15,
      page_title: 'pdf document foo',
      score: '2.1442184',
      span_id: '',
    },
    {
      page_id: 13,
      page_title: 'video',
      breadcrumbs: [],
      journal_about_page_id: 12,
      journal_id: 5,
      journal_name: 'new journal',
      block_id: 2,
      block_title: 'Science and Cooking Chef Profile: JOSÉ ANDRÉS',
      block_type: 'xblock_video',
      highlights: [
        'video-title: performance director <b>foo</b>',
      ],
      score: '1.2321665',
      span_id: 'xblock_video-c81e728d9d4c2f636f067f89cc14862c',
    },
    {
      page_id: 13,
      page_title: 'a page',
      breadcrumbs: [],
      journal_about_page_id: 12,
      journal_id: 5,
      journal_name: 'new journal',
      block_id: 1,
      block_title: 'image-title: foo',
      block_type: 'image',
      highlights: [
        'image-title: <b>foo</b>',
      ],
      score: '12.110976',
      span_id: 'image-c4ca4238a0b923820dcc509a6f75849b',
    },
  ],
};

const mockGetResultCount = jest.fn((filter) => {
  switch (filter) {
    case FILTER_ID_ALL:
      return mockSearchResults.hits.total_count;
    case FILTER_ID_DOCUMENTS:
      return mockSearchResults.hits.doc_count;
    case FILTER_ID_IMAGES:
      return mockSearchResults.hits.image_count;
    case FILTER_ID_VIDEOS:
      return mockSearchResults.hits.video_count;
    default:
      return 0;
  }
});

export { mockGetResultCount, mockSearchResults };

