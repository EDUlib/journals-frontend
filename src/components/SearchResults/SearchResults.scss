.results-container {
  margin-top: 20px;
}

.card-link {
  a:hover {
      text-decoration: none;
  }

  .search-result-card {
    display: block;
    padding: 10px 30px 0px 30px;
    text-align: left;
    border-radius: 4px;
    background-color: $white;
    border: solid 1px #d6dbe0;

    &:hover{
        background-color: $pale-grey;
        cursor: pointer;
    }
  }
}

.result-icon {
  font-size: 21px;
  line-height: 1.67;
  color: $gray-700;
}

.result-title {
  padding-left: 8px;
  font-family: "Open Sans";
  font-size: 20px;
  font-weight: 600;
  color: $deep-sea-blue;
  text-decoration: underline;
}

.result-highlight {
  font-family: "Open Sans";
  font-size: 16px;
  color: $dark-grey;
  display: block;
  padding-bottom: 10px;

  b {
    font-weight: bold;
    background-color:#dceef7;
    padding: 0px 4px 0px 4px;
    border-radius: 4px;
  }
}

.additional-highlights {
  font-family: "Open Sans";
  font-size: 16px;
  letter-spacing: normal;
  color: $slate-grey;
  padding-bottom: 10px;
}

.breadcrumb div{
  display: inline-block;
}

.result-breadcrumb {
  display: inline-block;
  font-family: "Open Sans";
  font-size: 14px;
  font-weight: 600;
  line-height: 1.71;
  letter-spacing: 1px;
  color: $slate-grey;
  padding-right: 4px;
  text-transform: uppercase
}

.result-breadcrumb:not(:last-child):after{
  content: ':';
}

.result-progress {
  margin: 20px;
  text-align: left;
}
