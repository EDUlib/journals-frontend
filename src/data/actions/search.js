import {
  STARTED_SEARCHING,
  FINISHED_SEARCHING,
  GET_SEARCH_SUCCESS,
  GET_SEARCH_FAILURE,
  SET_FILTER,
} from '../constants/actionTypes/search';

import JournalsApiService from '../services/JournalsApiService';


const startedSearching = (journalId, query, operator, filter) => (
  {
    type: STARTED_SEARCHING,
    journalId,
    query,
    operator,
    filter,
  }
);

const finishedSearching = () => (
  {
    type: FINISHED_SEARCHING,
  }
);

const getSearchSuccess = (meta, hits) => (
  {
    type: GET_SEARCH_SUCCESS,
    meta,
    hits,
  }
);

const getSearchFailure = error => (
  {
    type: GET_SEARCH_FAILURE,
    error,
  }
);

const setFilter = filter => (
  {
    type: SET_FILTER,
    filter,
  }
);

const fetchSearchResults = (journalId, query, operator, filter) => (
  (dispatch) => {
    dispatch(startedSearching(journalId, query, operator, filter));
    return JournalsApiService.fetchSearchResults(journalId, query, operator)
      .then((response) => {
        dispatch(getSearchSuccess(response.data.meta, response.data.hits));
      })
      .catch((error) => {
        dispatch(getSearchFailure(error));
      })
      .finally(() => {
        dispatch(finishedSearching());
      });
  }
);

export {
  setFilter,
  fetchSearchResults,
};
