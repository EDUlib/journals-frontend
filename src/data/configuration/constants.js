const settings = {
  journalsBackendBaseUrl: 'http://localhost:18606',
};

export default settings;
