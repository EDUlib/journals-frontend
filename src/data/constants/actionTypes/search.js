const STARTED_SEARCHING = 'STARTED_SEARCHING';
const FINISHED_SEARCHING = 'FINISHED_SEARCHING';
const GET_SEARCH_SUCCESS = 'GET_SEARCH_SUCCESS';
const GET_SEARCH_FAILURE = 'GET_SEARCH_FAILURE';
const SET_FILTER = 'SET_FILTER';

export {
  STARTED_SEARCHING,
  FINISHED_SEARCHING,
  GET_SEARCH_SUCCESS,
  GET_SEARCH_FAILURE,
  SET_FILTER,
};
