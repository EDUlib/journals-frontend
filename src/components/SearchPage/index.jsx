import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@edx/paragon';
import queryString from 'query-string';
import classNames from 'classnames';

import SearchResultsContainer from '../../containers/SearchResultsContainer';
import './SearchPage.scss';

import {
  FILTER_ID_ALL,
  FILTER_ID_IMAGES,
  FILTER_ID_DOCUMENTS,
  FILTER_ID_VIDEOS,
  FILTERS,
} from '../../data/constants/filterTypes';


class SearchPage extends React.Component {
  constructor(props) {
    super(props);
    this.getResultCount = this.getResultCount.bind(this);
  }


  componentDidMount() {
    const parsed = queryString.parse(this.props.location.search);

    if (!this.checkValidArgs(parsed)) {
      return;
    }

    this.props.getSearchResults(
      parseInt(parsed.journalId, 10),
      parsed.query,
      parsed.operator,
      this.props.filter,
    );
  }

  componentDidUpdate(prevProps) {
    const parsed = queryString.parse(this.props.location.search);

    if (!this.checkValidArgs(parsed)) {
      return;
    }

    const journalId = parseInt(parsed.journalId, 10);

    if (!this.props.error && !this.props.searchStarted &&
      (prevProps.query !== parsed.query ||
      prevProps.operator !== parsed.operator ||
      prevProps.journalId !== journalId)) {
      this.props.getSearchResults(
        journalId,
        parsed.query,
        parsed.operator,
        this.props.filter,
      );
    }
  }

  getResultCount(filter) {
    if (this.props.meta.total_count === undefined) {
      return 0;
    }

    switch (filter) {
      case FILTER_ID_ALL:
        return this.props.meta.total_count;
      case FILTER_ID_DOCUMENTS:
        return this.props.meta.doc_count;
      case FILTER_ID_IMAGES:
        return this.props.meta.image_count;
      case FILTER_ID_VIDEOS:
        return this.props.meta.video_count;
      default:
        return 0;
    }
  }

  handleClick(filter) {
    this.props.setFilter(filter);
  }

  checkValidArgs(parsed) {
    if (parsed.journalId === undefined ||
      parsed.query === undefined ||
      parsed.operator === undefined) {
      return false;
    }
    return true;
  }

  render() {
    return (
      <div>
        <div className="search-filter-panel">
          {
            Object.keys(FILTERS).map(label => (
              <div key={label}>
                <Button
                  className={
                      classNames({
                        'filter-btn': true,
                        'filter-btn-selected': FILTERS[label] === this.props.filter,
                      }).split(' ')
                    }
                  label={`${label} (${this.getResultCount(FILTERS[label])})`}
                  onClick={() => this.handleClick(FILTERS[label])}
                />
              </div>
            ))
        }
        </div>
        <SearchResultsContainer resultsCounter={this.getResultCount} />
      </div>
    );
  }
}


SearchPage.defaultProps = {
  setFilter: () => {},
  journalId: 0,
  getSearchResults: () => [{}],
  searchStarted: false,
  location: {
    search: null,
  },
  error: null,
};

SearchPage.propTypes = {
  filter: PropTypes.string.isRequired,
  meta: PropTypes.shape({
    total_count: PropTypes.number,
    text_count: PropTypes.number,
    image_count: PropTypes.number,
    video_count: PropTypes.number,
    doc_count: PropTypes.number,
  }).isRequired,
  journalId: PropTypes.number,
  query: PropTypes.string.isRequired,
  operator: PropTypes.string.isRequired,
  getSearchResults: PropTypes.func,
  location: PropTypes.shape({
    search: PropTypes.string,
  }),
  setFilter: PropTypes.func,
  searchStarted: PropTypes.bool,
  error: PropTypes.instanceOf(Error),
};

export default SearchPage;
