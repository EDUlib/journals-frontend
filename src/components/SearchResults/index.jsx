import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { HashLink as Link } from 'react-router-hash-link';
import { Icon } from '@edx/paragon';
import { FormattedPlural } from 'react-intl';

import './SearchResults.scss';
import {
  RICH_TEXT,
  RAW_HTML,
  PDF,
  XBLOCK_VIDEO,
  IMAGE,
} from '../../data/constants/contentTypes';

import {
  FILTER_ID_ALL,
  FILTER_ID_IMAGES,
  FILTER_ID_DOCUMENTS,
  FILTER_ID_VIDEOS,
} from '../../data/constants/filterTypes';

const NUM_HIGHLIGHTS_DISPLAY = 3;

const BreadCrumb = ({
  breadcrumbs,
}) => (
  <div className="breadcrumb">
    {
    breadcrumbs.map(breadcrumb => (
      <div key={breadcrumb} className="result-breadcrumb">
        {breadcrumb}
      </div>
    ))
  }
  </div>
);

const AdditionalHighlights = ({
  total,
}) => (
  total > NUM_HIGHLIGHTS_DISPLAY &&
    <div className="additional-highlights">
      <FormattedPlural
        value={total - NUM_HIGHLIGHTS_DISPLAY}
        one={`+ ${total - NUM_HIGHLIGHTS_DISPLAY} more instance`}
        other={`+ ${total - NUM_HIGHLIGHTS_DISPLAY} more instances`}
      />
    </div>
);

const SearchHighlight = ({
  highlights,
}) => (
  highlights.map(highlight => (
    <div
      key={highlight}
      className="result-highlight"
      dangerouslySetInnerHTML={{ __html: highlight }} // eslint-disable-line react/no-danger
    />
  ))
);

const SearchItem = ({
  data,
}) => (
  <Link className="card-link" to={`/${data.journal_about_page_id}/pages/${data.page_id}#${data.span_id}`} >
    <div className="search-result-card">
      <div className="result-icon">
        <Icon
          className={
            classNames({
              fa: true,
              'fa-book': data.block_type === RICH_TEXT || data.block_type === RAW_HTML,
              'fa-file-video-o': data.block_type === XBLOCK_VIDEO,
              'fa-picture-o': data.block_type === IMAGE,
              'fa-file-pdf-o': data.block_type === PDF,
            }).split(' ')
          }
        />
        <span className="result-title">{data.block_title}</span>
      </div>

      <SearchHighlight highlights={data.highlights.slice(0, NUM_HIGHLIGHTS_DISPLAY)} />
      <AdditionalHighlights total={data.highlights.length} />
      <BreadCrumb breadcrumbs={data.breadcrumbs} />
    </div>
  </Link>
);

class SearchResults extends React.Component {
  formatKey(searchResults) {
    return `${searchResults.journal_id}-${searchResults.page_id}-${searchResults.block_id}-${searchResults.block_type}`;
  }

  matchesFilter(blockType) {
    return (this.props.filter === FILTER_ID_ALL ||
            (this.props.filter === FILTER_ID_DOCUMENTS && blockType === PDF) ||
              (this.props.filter === FILTER_ID_IMAGES && blockType === IMAGE) ||
              (this.props.filter === FILTER_ID_VIDEOS && blockType === XBLOCK_VIDEO));
  }

  render() {
    if (this.props.error) {
      return (
        <div className="result-progress"> Error occurred searching</div>
      );
    }
    if (this.props.searchStarted && !this.props.searchFinished) {
      return (
        <div className="result-progress"> Searching...</div>
      );
    }
    if (this.props.resultsCounter(this.props.filter) === 0) {
      return (
        <div className="result-progress">No matching results found</div>
      );
    }
    return (
      <div>
        <div className="results-container">
          {
            this.props.hits.map(searchResult => (
              this.matchesFilter(searchResult.block_type) &&
                <SearchItem
                  key={this.formatKey(searchResult)}
                  data={searchResult}
                  filter={this.props.filter}
                />
            ))
          }
        </div>
      </div>
    );
  }
}

BreadCrumb.defaultProps = {
  breadcrumbs: [],
};

BreadCrumb.propTypes = {
  breadcrumbs: PropTypes.arrayOf(PropTypes.string),
};

SearchItem.defaultProps = {
};

SearchItem.propTypes = {
  data: PropTypes.shape({
    journal_about_page_id: PropTypes.number,
    page_id: PropTypes.number,
    span_id: PropTypes.string,
    block_title: PropTypes.string,
    block_type: PropTypes.string,
    highlights: PropTypes.array,
    breadcrumbs: PropTypes.array,
  }).isRequired,
};

SearchResults.defaultProps = {
  hits: [],
  searchFinished: true,
  searchStarted: false,
  error: null,
  resultsCounter: () => 0,
};

SearchResults.propTypes = {
  resultsCounter: PropTypes.func,
  filter: PropTypes.string.isRequired,
  searchFinished: PropTypes.bool,
  searchStarted: PropTypes.bool,
  error: PropTypes.instanceOf(Error),
  hits: PropTypes.arrayOf(PropTypes.object),
};

export default SearchResults;
