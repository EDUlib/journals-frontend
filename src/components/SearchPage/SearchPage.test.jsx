import React from 'react';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import configureMockStore from 'redux-mock-store';
import { FILTER_ID_ALL, FILTER_ID_DOCUMENTS } from '../../data/constants/filterTypes';
import { SET_FILTER, STARTED_SEARCHING } from '../../data/constants/actionTypes/search';

import { mockSearchResults } from '../SearchResults/SearchResults.mock';
import SearchPage from './index';

const mockStore = configureMockStore([thunk]);

const searchResults = {
  filter: FILTER_ID_ALL,
  hits: mockSearchResults.hits,
  error: null,
  finishedFetching: true,
  startedFetching: false,
};
const store = mockStore({ searchResults });

const mockSetFilter = (filter) => {
  store.dispatch({ type: SET_FILTER, filter });
};
const mockGetSearchResults = (journalId, query, operator, filter) => {
  store.dispatch({
    type: STARTED_SEARCHING,
    journalId,
    query,
    operator,
    filter,
  });
};

const testActionPayload = (expectedPayload) => {
  const actions = store.getActions();
  if (expectedPayload) {
    expect(actions).toEqual([expectedPayload]);
  } else {
    expect(actions).toEqual([]);
  }
};

const SearchPageWrapper = props => (
  <IntlProvider locale="en">
    <MemoryRouter>
      <Provider store={store}>
        <SearchPage
          {...props}
        />
      </Provider>
    </MemoryRouter>
  </IntlProvider>
);

describe('<SearchPage/>', () => {
  beforeEach(() => {
    store.clearActions();
  });
  describe('renders', () => {
    it('with all filters', () => {
      const tree = renderer
        .create((
          <SearchPageWrapper filter={FILTER_ID_ALL} query="foo" operator="and" meta={mockSearchResults.meta} />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with empty meta', () => {
      const wrapper = mount(<SearchPageWrapper
        filter={FILTER_ID_ALL}
        query="foo"
        operator="and"
        meta={{}}
      />);
      const cardLength = wrapper.find('.search-result-card').length;
      expect(cardLength).toBe(0);
    });

    it('with invalid filter', () => {
      const wrapper = mount(<SearchPageWrapper
        filter="blah"
        query="foo"
        operator="and"
        meta={mockSearchResults.meta}
      />);
      expect(wrapper.find('.filter-btn-selected').length).toBe(0);
    });

    it('renders correct results', () => {
      const wrapper = mount(<SearchPageWrapper
        filter={FILTER_ID_ALL}
        query="foo"
        operator="and"
        meta={mockSearchResults.meta}
      />);
      const cardLength = wrapper.find('.search-result-card').length;
      expect(cardLength).toBe(mockSearchResults.meta.total_count);
      expect(wrapper.find('.filter-btn-selected').length).toBe(1);
    });
  });

  describe('dispatch', () => {
    it('change filter correctly', () => {
      const wrapper = mount(<SearchPageWrapper
        setFilter={mockSetFilter}
        filter={FILTER_ID_ALL}
        query="foo"
        operator="and"
        meta={mockSearchResults.meta}
      />);
      wrapper.find('button.filter-btn').at(1).simulate('click');
      const actions = store.getActions();
      const expectedPayload = { type: SET_FILTER, filter: FILTER_ID_DOCUMENTS };
      expect(actions).toEqual([expectedPayload]);
    });
    describe('searches', () => {
      it('correctly', () => {
        let props = {
          journalId: 1,
          query: 'foo',
          operator: 'or',
          filter: FILTER_ID_ALL,
          getSearchResults: mockGetSearchResults,
        };
        props = {
          ...props,
          location: { search: `journalId=${props.journalId}&operator=${props.operator}&query=${props.query}` },
        };
        const wrapper = mount(<SearchPageWrapper {...props} meta={mockSearchResults.meta} />);
        store.clearActions();
        const NewProps = {
          journalId: 2,
          query: 'bar',
          operator: 'and',
          filter: props.filter,
        };
        const location = { search: `journalId=${NewProps.journalId}&operator=${NewProps.operator}&query=${NewProps.query}` };
        wrapper.setProps({ NewProps, location });
        testActionPayload({ type: STARTED_SEARCHING, ...NewProps });
      });

      it('undefined journalId', () => {
        let props = {
          query: 'foo',
          operator: 'or',
          filter: FILTER_ID_ALL,
          getSearchResults: mockGetSearchResults,
        };
        props = {
          ...props,
          location: { search: `operator=${props.operator}&query=${props.query}` },
        };
        const wrapper = mount(<SearchPageWrapper {...props} meta={mockSearchResults.meta} />);
        testActionPayload(null);

        // try searching again with different query to update component
        wrapper.setProps({ query: 'bar' });
        testActionPayload(null);
      });

      it('without passing getSearchResults', () => {
        let props = {
          query: 'foo',
          operator: 'or',
          filter: FILTER_ID_ALL,
        };
        props = {
          ...props,
          location: { search: `journalId=${props.journalId}&operator=${props.operator}&query=${props.query}` },
        };
        mount(<SearchPageWrapper {...props} meta={mockSearchResults.meta} />);
        testActionPayload(null);
      });

      it('with error', () => {
        let props = {
          journalId: 1,
          query: 'foo',
          operator: 'or',
          filter: FILTER_ID_ALL,
        };
        props = {
          ...props,
          location: { search: `journalId=${props.journalId}&operator=${props.operator}&query=${props.query}` },
        };
        const wrapper = mount(<SearchPageWrapper {...props} meta={mockSearchResults.meta} />);
        // try searching again with different query to update component
        wrapper.setProps({ query: 'bar', error: new Error() });
        testActionPayload(null);
      });

      it('change journal id', () => {
        let props = {
          journalId: 1,
          query: 'foo',
          operator: 'or',
          filter: FILTER_ID_ALL,
        };
        props = {
          ...props,
          location: { search: `journalId=${props.journalId}&operator=${props.operator}&query=${props.query}` },
        };
        const wrapper = mount(<SearchPageWrapper {...props} meta={mockSearchResults.meta} />);
        // try searching again with different query to update component
        const newJournalId = 2;
        const location = { search: `journalId=${newJournalId}&operator=${props.operator}&query=${props.query}` };
        wrapper.setProps({ journalId: newJournalId, location });
        testActionPayload(null);
      });
    });
  });
  describe('getSearchResult', () => {
    const wrapper = mount(<SearchPageWrapper
      filter={FILTER_ID_ALL}
      query="foo"
      operator="and"
      meta={mockSearchResults.meta}
    />);
    const instance = wrapper.find('SearchPage').instance();
    it('with filter all', () => {
      expect(instance.getResultCount(FILTER_ID_ALL)).toBe(mockSearchResults.meta.total_count);
    });

    it('with document filter', () => {
      expect(instance.getResultCount(FILTER_ID_DOCUMENTS)).toBe(mockSearchResults.meta.doc_count);
    });

    it('with invalid filter', () => {
      expect(instance.getResultCount('blah')).toBe(0);
    });
  });
});
