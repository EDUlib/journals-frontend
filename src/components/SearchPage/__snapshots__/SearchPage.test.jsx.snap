// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage/> renders with all filters 1`] = `
<div>
  <div
    className="search-filter-panel"
  >
    <div>
      <button
        className="btn filter-btn filter-btn-selected"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        All Results (3)
      </button>
    </div>
    <div>
      <button
        className="btn filter-btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Documents (1)
      </button>
    </div>
    <div>
      <button
        className="btn filter-btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Images (1)
      </button>
    </div>
    <div>
      <button
        className="btn filter-btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Videos (1)
      </button>
    </div>
  </div>
  <div>
    <div
      className="results-container"
    >
      <a
        className="card-link"
        href="/12/pages/15#"
        onClick={[Function]}
      >
        <div
          className="search-result-card"
        >
          <div
            className="result-icon"
          >
            <span
              aria-hidden={true}
              className="fa fa-file-pdf-o"
              id="Icon1"
            />
            <span
              className="result-title"
            >
              block 1
            </span>
          </div>
          <div
            className="result-highlight"
            dangerouslySetInnerHTML={
              Object {
                "__html": "1 <b>foo </b>",
              }
            }
          />
          <div
            className="result-highlight"
            dangerouslySetInnerHTML={
              Object {
                "__html": "2 <b>foo </b>",
              }
            }
          />
          <div
            className="result-highlight"
            dangerouslySetInnerHTML={
              Object {
                "__html": "3 <b>foo </b>",
              }
            }
          />
          <div
            className="additional-highlights"
          >
            <span>
              + 2 more instances
            </span>
          </div>
          <div
            className="breadcrumb"
          >
            <div
              className="result-breadcrumb"
            >
              x
            </div>
            <div
              className="result-breadcrumb"
            >
              y
            </div>
          </div>
        </div>
      </a>
      <a
        className="card-link"
        href="/12/pages/13#xblock_video-c81e728d9d4c2f636f067f89cc14862c"
        onClick={[Function]}
      >
        <div
          className="search-result-card"
        >
          <div
            className="result-icon"
          >
            <span
              aria-hidden={true}
              className="fa fa-file-video-o"
              id="Icon1"
            />
            <span
              className="result-title"
            >
              Science and Cooking Chef Profile: JOSÉ ANDRÉS
            </span>
          </div>
          <div
            className="result-highlight"
            dangerouslySetInnerHTML={
              Object {
                "__html": "video-title: performance director <b>foo</b>",
              }
            }
          />
          <div
            className="breadcrumb"
          />
        </div>
      </a>
      <a
        className="card-link"
        href="/12/pages/13#image-c4ca4238a0b923820dcc509a6f75849b"
        onClick={[Function]}
      >
        <div
          className="search-result-card"
        >
          <div
            className="result-icon"
          >
            <span
              aria-hidden={true}
              className="fa fa-picture-o"
              id="Icon1"
            />
            <span
              className="result-title"
            >
              image-title: foo
            </span>
          </div>
          <div
            className="result-highlight"
            dangerouslySetInnerHTML={
              Object {
                "__html": "image-title: <b>foo</b>",
              }
            }
          />
          <div
            className="breadcrumb"
          />
        </div>
      </a>
    </div>
  </div>
</div>
`;
